services/src/main/java/org/exoplatform/news/service/impl/NewsTargetingServiceImpl.java

```diff
@@ -70,13 +76,35 @@ public NewsTargetingServiceImpl(MetadataService metadataService, IdentityManager
   }
 
   @Override
-  public List<NewsTargetingEntity> getTargets() {
+  public List<NewsTargetingEntity> getAllTargets() {
     List<Metadata> targets = metadataService.getMetadatas(METADATA_TYPE.getName(), LIMIT);
     return targets.stream().map(this::toEntity).collect(Collectors.toList());
   }
   
   @Override
-  public void deleteTargetByName(String targetName, org.exoplatform.services.security.Identity  currentIdentity) throws IllegalAccessException {
+  public List<NewsTargetingEntity> getAllowedTargets(org.exoplatform.services.security.Identity userIdentity) {
+    List<Metadata> allTargetsMetadatas = metadataService.getMetadatas(METADATA_TYPE.getName(), LIMIT);
+    List<Metadata> allowedTargets = new ArrayList<>();
+    for (Metadata targetMetadata : allTargetsMetadatas) {
+      List<String> targetMetadataPermissions = List.of(targetMetadata.getProperties().get(NewsUtils.TARGET_PERMISSIONS).split(","));
+      for (String targetMetadataPermission : targetMetadataPermissions) {
+        if (targetMetadataPermission.contains(SPACE_TARGET_PERMISSION_PREFIX)) {
+          Space space = spaceService.getSpaceByDisplayName(targetMetadataPermission.replace(SPACE_TARGET_PERMISSION_PREFIX, ""));
```

Comment:
for the target space permission;, we should store the spaceId instead of spacePrettyName which will avoid problems when renaming the space since space pretty name will be changed but space id will not be change